static/js/narrow_state.js

```diff
@@ -73,7 +73,7 @@ function collect_single(operators) {
 // This logic is here and not in the 'compose' module because
 // it will get more complicated as we add things to the narrow
 // operator language.
-exports.set_compose_defaults = function (opts) {
+exports.set_compose_defaults = function (opts = {}) {
```

Comment:
We probably don't want a default value here, since that makes the function still be mutable.

I would say `var opts = {};` in the first line, and you'll have to change the caller in `compose_actions.js`.  Does that make sense?

Reply:
Also, this change seems unrelated the commit message.  You want to make sure each commit is "atomic" and makes sense on its own, without doing too many things at once.

Reply:
The commit message is because of GCI,

Reply:
This commit message isn't related to the function that you are trying to cover.

Reply:
@showell what do you mean by
> and you'll have to change the caller in compose_actions.js

(I we remove the support for mutating the object passed the function `fill_in_opts_from_current_narrowed_view` will not run as expected? )